store/tikv/2pc.go

```diff
@@ -1523,17 +1523,44 @@ func (c *twoPhaseCommitter) tryAmendTxn(ctx context.Context, startInfoSchema Sch
 		c.doingAmend = true
 		defer func() { c.doingAmend = false }()
 		if keysNeedToLock.len() > 0 {
-			pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
 			lCtx := &kv.LockCtx{
-				ForUpdateTS:  c.forUpdateTS,
-				LockWaitTime: kv.LockNoWait,
+				Killed:        c.lockCtx.Killed,
+				ForUpdateTS:   c.forUpdateTS,
+				LockWaitTime:  c.lockCtx.LockWaitTime,
+				WaitStartTime: time.Now(),
+			}
+			tryTimes := uint(0)
+			retryLimit := config.GetGlobalConfig().PessimisticTxn.MaxRetryCount
+			for tryTimes < retryLimit {
+				pessimisticLockBo := NewBackofferWithVars(ctx, pessimisticLockMaxBackoff, c.txn.vars)
+				err = c.pessimisticLockMutations(pessimisticLockBo, lCtx, keysNeedToLock)
+				if err != nil {
+					// KeysNeedToLock won't change, so don't async rollback pessimistic locks here for write conflict.
+					if terror.ErrorEqual(kv.ErrWriteConflict, err) {
+						newForUpdateTSVer, err := c.store.CurrentVersion()
+						if err != nil {
+							return false, errors.Trace(err)
```

Comment:
Do we need to add a warning log before the return?

Reply:
It rarely happens and the error message will be printed outside this function call.